elizaOS/eliza
# feat: add theme toggle functionality with dark and light mode support

- Imported ThemeToggle component in AppSidebar
- Added ThemeToggle to SidebarFooter in AppSidebar
- Initialized theme based on localStorage and system preferences in main.tsx to support dark and light modes

# Relates to:



# Risks

Low. This change primarily affects the UI theme settings and should not impact other functionalities.

# Background

## What does this PR do?

This PR adds a theme toggle functionality to the application, allowing users to switch between dark and light modes. It also initializes the theme based on the user's system preferences and previously saved settings in localStorage.

## What kind of change is this?

Features (non-breaking change which adds functionality)

# Documentation changes needed?

My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?

Review the changes in `client/src/components/app-sidebar.tsx` and `client/src/main.tsx`.

## Detailed testing steps

1. Run the application.
2. Verify that the theme toggle appears in the sidebar footer.
3. Test switching between dark and light modes using the theme toggle.
4. Refresh the page and ensure the selected theme persists.
5. Change the system theme and verify the application updates accordingly when set to "system" mode.
